server/v2/api/grpc/gogoreflection/serverreflection.go

```diff
@@ -458,26 +458,60 @@ func (s *serverReflectionServer) ServerReflectionInfo(stream rpb.ServerReflectio
 }
 
 // getServices gets the unique list of services given a list of methods.
-func (s *serverReflectionServer) getServices(methods []string) (svcs []string, fds []*dpb.FileDescriptorProto) {
+func (s *serverReflectionServer) getServices(messages []string) (svcs []string, fds []*dpb.FileDescriptorProto) {
 	registry, err := gogoproto.MergedRegistry()
 	if err != nil {
 		s.log.Error("unable to load merged registry", "err", err)
 		return nil, nil
 	}
 	seenSvc := map[protoreflect.FullName]struct{}{}
-	for _, methodName := range methods {
-		methodName = strings.Join(strings.Split(methodName[1:], "/"), ".")
-		md, err := registry.FindDescriptorByName(protoreflect.FullName(methodName))
+	for _, messageName := range messages {
+		md, err := registry.FindDescriptorByName(protoreflect.FullName(messageName))
 		if err != nil {
-			s.log.Error("unable to load method descriptor", "method", methodName, "err", err)
+			s.log.Error("unable to load message descriptor", "message", messageName, "err", err)
 			continue
 		}
-		svc := md.(protoreflect.MethodDescriptor).Parent()
+
+		svc, ok := findServiceForMessage(registry, md.(protoreflect.MessageDescriptor))
+		if !ok {
+			s.log.Warn(fmt.Sprintf("unable to find service for message %s", messageName))
```

Comment:
I think this should be error level. 
Also, can we extend the log and add instruction what a developer should do to resolve the problem? Often the errors are hard to resolve without a bit more context

Reply:
Actually there may be valid cases where this would always error. This is why I decreaded from error to warn.

For instance, appmodulev2 handlers (the alternative way to wire msgs without grpc in v2), doesn't make use of grpc services. Meaning this would always log an error.

I almost feel like it should be a debug level log, as afaik that would be this only case this can happen.